fs/operations.go

```diff
@@ -1303,3 +1364,24 @@ func MoveFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err err
 func CopyFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err error) {
 	return moveOrCopyFile(fdst, fsrc, dstFileName, srcFileName, true)
 }
+
+func (j jsonPrinter) Start(w io.Writer) {
+	syncFprintf(w, "[")
+}
+
+func (j jsonPrinter) End(w io.Writer) {
+	syncFprintf(w, "]")
+}
+
+func (j *jsonPrinter) Fprintf(w io.Writer, skipSeperator bool, format string, a ...interface{}) {
+	// Lets skip the seperator when calling the function itself
+	if j.seperatorRequired && !skipSeperator {
+		j.Fprintf(w, true, ",")
+	}
+	// Set the seperator before locking so parallel things don't get in the way of the lock
+	j.seperatorRequired = true
+
+	j.mu.Lock()
```

Comment:
Maybe it could be that when the if block is satisfied (at the start of the method), it is recursively calling the method and for some reason the lock isn't being released?

I tried to break/get stacktraces from the process but it wasn't accepting anything. Ctrl+C and Ctrl+Break closes it, but that's it.